tesseract_process_managers/include/tesseract_process_managers/taskflow_generators/graph_taskflow.h

```diff
@@ -48,6 +48,11 @@ class GraphTaskflow : public TaskflowGenerator
 public:
   using UPtr = std::unique_ptr<GraphTaskflow>;
 
+  /** @brief ID for the Done node */
+  const static int DONE_NODE = -1;
+  /** @brief ID for the Error node */
+  const static int ERROR_NODE = -2;
```

Comment:
```suggestion
  /** @brief ID for the Done node */
  const static int DONE_NODE = 0;
  /** @brief ID for the Error node */
  const static int ERROR_NODE = 1;
```
It's probably easier for these nodes to be valid vector indices so we can index into `nodes_` without doing a bunch of special checks about whether the node index is less than zero and whether it equals one of these two numbers

Reply:
The error and done tasks are not currently added to nodes, so this allows you to index into nodes_ without subtracting 2 every time. The only check you have to do is if it is less than 0. Those cases are already handled in an ad hoc fashion, so I don't see a couple of if statements as much worse than that.

Reply:
I forgot the error and done tasks are added after all of the other nodes have been added. Keeping the negative indices probably make sense then. The alternative would be adding two placeholder nodes to the `nodes_` vector, then replacing them when the actual error and done tasks get generated. That seems relatively hacky though